std/array.d

```diff
@@ -482,6 +482,37 @@ auto byPair(Key, Value)(Value[Key] aa)
     assert(savedPairs.front == tuple("a", 2));
 }
 
+// Issue 17711
+@system unittest
+{
+    const(int[string]) aa = [ "abc": 123 ];
+
+    // Ensure that byKeyValue is usable with a const AA.
+    auto kv = aa.byKeyValue;
+    assert(!kv.empty);
+    assert(kv.front.key == "abc" && kv.front.value == 123);
+    kv.popFront();
+    assert(kv.empty);
+
+    // Ensure byPair is instantiable with const AA.
+    auto r = aa.byPair;
+    static assert(isInputRange!(typeof(r)));
+    assert(!r.empty && r.front[0] == "abc" && r.front[1] == 123);
+    r.popFront();
+    assert(r.empty);
+
+    // Original failing user code (simplified)
+    class A
+    {
+        this() { aa = ["a":1, "b" : 2, "c" : 3]; }
+        auto pairs() @property const { return aa.byPair.array; }
+    private:
+         int[string] aa;
+    }
+    A a;
+    static assert(is(typeof(a)));
```

Comment:
You mean the entire class definition?  I'd tend to agree... I just thought it was extra insurance in case something else about the user's code may cause things to break.